src/main/datatypes/MQ2WindowType.cpp

```diff
@@ -860,4 +908,156 @@ bool MQ2WindowType::dataWindow(const char* szIndex, MQTypeVar& Ret)
+		strcpy_s(DataTypeTemp, pInvWnd->pBackground->GetName().c_str());
+		Dest.Ptr = &DataTypeTemp[0];
+		return true;
+
+
+	case InvSlotWindowMembers::ItemLocation:
+		Dest.Type = pIntType;
+		Dest.DWord = pInvWnd->ItemLocation.GetLocation();
+		return true;
+
+	case InvSlotWindowMembers::ItemIndexSlot:
+		if (IsNumber(Index))
+		{
+			int nIndex = GetIntFromString(Index, 0);
+			Dest.Type = pIntType;
+			Dest.DWord = pInvWnd->ItemLocation.GetIndex().GetSlot(nIndex);
+			return true;
+		}
+		return false;
+
+
+	case InvSlotWindowMembers::Item:
+		Dest.Type = pItemType;
+		Dest = pItemType->MakeTypeVar(pInvWnd->LinkedItem ? pInvWnd->LinkedItem : pLocalPC->GetItemByGlobalIndex(pInvWnd->ItemLocation));
+		return true;
+
+	case InvSlotWindowMembers::ItemOffsetX:
+		Dest.Type = pIntType;
+		Dest.DWord = pInvWnd->ItemOffsetX;
+		return true;
+
+	case InvSlotWindowMembers::ItemOffsetY:
+		Dest.Type = pIntType;
+		Dest.DWord = pInvWnd->ItemOffsetY;
+		return true;
+
+	case InvSlotWindowMembers::ItemTexture:
+		Dest.Type = pStringType;
+		strcpy_s(DataTypeTemp, pInvWnd->ptItem->GetName().c_str());
+		Dest.Ptr = &DataTypeTemp[0];
+		return true;
+
+	case InvSlotWindowMembers::Quantity:
+		Dest.Type = pIntType;
+		Dest.DWord = pInvWnd->Quantity;
+		return true;
+
+	case InvSlotWindowMembers::Selected:
+		Dest.Type = pBoolType;
+		Dest.DWord = pInvWnd->bSelected;
+		return true;
+
+
+	case InvSlotWindowMembers::FindSelected:
+		Dest.Type = pBoolType;
+		Dest.DWord = pInvWnd->bFindSelected;
+		return true;
+
+	case InvSlotWindowMembers::HotButton:
+		Dest.Type = pBoolType;
+		Dest.DWord = pInvWnd->bHotButton;
+		return true;
+
+	case InvSlotWindowMembers::InventorySlotLinked:
+		Dest.Type = pBoolType;
+		Dest.DWord = pInvWnd->bInventorySlotLinked;
+		return true;
+
+	case InvSlotWindowMembers::Text:
+		Dest.Type = pStringType;
+		strcpy_s(DataTypeTemp, pInvWnd->pTextObject->GetText().c_str());
+		Dest.Ptr = &DataTypeTemp[0];
+		return true;
+
+
+	case InvSlotWindowMembers::Mode:
+		Dest.Type = pIntType;
+		Dest.DWord = pInvWnd->Mode;
+		return true;
+
+	}
+	return false;
+}
+
+bool MQ2InvSlotWindowType::ToString(MQVarPtr VarPtr, char* Destination)
+{
+	CXWnd* pWnd = static_cast<CXWnd*>(VarPtr.Ptr);
+
+	if (pWnd)
+	{
+		if (pWnd->IsVisible())
+			strcpy_s(Destination, MAX_STRING, "TRUE");
+		else
+			strcpy_s(Destination, MAX_STRING, "FALSE");
+	}
+	return true;
+}
+
+bool MQ2InvSlotWindowType::FromData(MQVarPtr& VarPtr, const MQTypeVar& Source)
+{
+	if (Source.Type != pInvSlotWindowType)
+		return false;
+
+	VarPtr.Ptr = Source.Ptr;
+	return true;
+}
+
+bool MQ2InvSlotWindowType::FromString(MQVarPtr& VarPtr, const char* Source)
+{
+	if (VarPtr.Ptr = FindMQ2WindowPath(Source))
+		return true;
+
+	return false;
+}
+
+bool MQ2InvSlotWindowType::dataWindow(const char* szIndex, MQTypeVar& Ret)
+{
+	if (szIndex[0])
+	{
+		if (Ret.Ptr = FindMQ2Window(szIndex))
+		{
+			Ret.Type = pInvSlotWindowType;
+			return true;
+		}
+	}
+
+	return false;
+}
```

Comment:
you don't want this function, it's what's creating the TLO (and this new type doesn't need a TLO)

Reply:
done.

Reply:
done